DriverDataTimelineUI/src/main/java/com/drivequant/drivekit/timeline/ui/timeline/TimelineViewModel.kt

```diff
@@ -84,84 +74,62 @@ internal class TimelineViewModel(application: Application) : AndroidViewModel(ap
                 update()
             }
         }
-        DriveKitDriverData.getRawTimelines(
-            this.periods,
-            object : TimelineQueryListener {
-                override fun onResponse(
-                    timelineSyncStatus: TimelineSyncStatus,
-                    timelines: List<DKRawTimeline>
-                ) {
-                    if (timelineSyncStatus == TimelineSyncStatus.CACHE_DATA_ONLY) {
-                        timelines.forEach {
-                            when (it.period) {
-                                DKPeriod.WEEK -> weekTimeline = it
-                                DKPeriod.MONTH -> monthTimeline = it
-                                DKPeriod.YEAR -> {}
-                            }
-                        }
-                        update()
-                    }
+        DriveKitDriverData.getDriverTimelines(this.periods, SynchronizationType.CACHE, true) { timelineSyncStatus, timelines ->
+            if (timelineSyncStatus == TimelineSyncStatus.CACHE_DATA_ONLY) {
+                timelines.forEach {
+                    timelineByPeriod[it.period] = it
                 }
-            },
-            SynchronizationType.CACHE
-        )
+                update()
+            }
+        }
         updateTimeline()
     }
 
     fun updateTimeline() {
-        DriveKitDriverData.getRawTimelines(
-            this.periods,
-            object : TimelineQueryListener {
-                override fun onResponse(
-                    timelineSyncStatus: TimelineSyncStatus,
-                    timelines: List<DKRawTimeline>
-                ) {
-                    if (timelineSyncStatus != TimelineSyncStatus.NO_TIMELINE_YET) {
-                        timelines.forEach {
-                            when (it.period) {
-                                DKPeriod.WEEK -> weekTimeline = it
-                                DKPeriod.MONTH -> monthTimeline = it
-                                DKPeriod.YEAR -> {}
-                            }
-                        }
-                        update(resettingSelectedDate = true)
-                    }
-                    syncStatus.postValue(timelineSyncStatus)
+        DriveKitDriverData.getDriverTimelines(this.periods, ignoreItemsWithoutTripScored = true) { timelineSyncStatus, timelines ->
+            if (timelineSyncStatus != TimelineSyncStatus.NO_TIMELINE_YET) {
+                timelines.forEach {
+                    timelineByPeriod[it.period] = it
                 }
```

Comment:
Pareil que https://github.com/DriveQuantPublic/drivekit-ui-android/pull/369/files#r1525154438
```suggestion
                timelines.forEach {
                    timelineByPeriod = timelines.associateBy { it.period }
                }
```

Reply:
Je me suis trompé dans la suggestion désolé 😓